### PR TITLE
Bump pexpect to 4.9

### PR DESCRIPTION
**Pull request by tapple**
The only api change is the addition of SocketSpawn, which I also contributed: https://github.com/pexpect/pexpect/pull/745

I also fixed some Incomplete typings while I was at it

### CODE REVIEW

File: stubs/pexpect/pexpect/fdpexpect.pyi [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
-    name: Incomplete
-    use_poll: Incomplete
+    name: str
+    use_poll: bool
     def __init__(
         self,
-        fd,
-        args: Incomplete | None = None,
-        timeout: int = 30,
+        fd: int,
```
Comment (fperrin) [inline stubs/pexpect/pexpect/fdpexpect.pyi:18]: Hi @tapple 

I was about to do something similar, you beat me to the punch :) (and you went further than I did).

`fd` should be [`FileDescriptorLike`](https://github.com/python/typeshed/blob/3dbb2573e68a288ac5ad1b4967a9c5ab994b81be/stdlib/_typeshed/__init__.pyi#L236), as the constructor accepts any object with a `fileno()` method (in addition to int).

File: stubs/pexpect/pexpect/spawnbase.pyi [hunk 2]
```diff
@@ -16,66 +21,96 @@ class _Logfile(Protocol):
...
-    status: Incomplete
+    exitstatus: int | None
+    signalstatus: int | None
+    status: int | None
     child_fd: int
-    timeout: Incomplete
-    delimiter: Incomplete
+    timeout: float | None
+    delimiter: type[EOF]
     logfile: _Logfile
```
Comment (fperrin) [inline stubs/pexpect/pexpect/spawnbase.pyi:50]: While you're making changes here, could you change `logfile` and friends to `_Logfile | None`?

**Comment by tapple** [comment]
I just saw your review comments. gonna work on those today

File: stubs/pexpect/pexpect/spawnbase.pyi (hunk 2 above)
Comment (tapple) [reply stubs/pexpect/pexpect/spawnbase.pyi]: fixed

File: stubs/pexpect/pexpect/fdpexpect.pyi (hunk 1 above)
Comment (tapple) [reply stubs/pexpect/pexpect/fdpexpect.pyi]: fixed

**Comment by tapple** [comment]
@fperrin I addressed your comments

**Review by hauntsaninja — APPROVED** [review]
Thank you!

**Pull request merged**